src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Stream.cs

```diff
@@ -106,6 +108,10 @@ public Http2Stream(HttpRequestMessage request, Http2Connection connection)
                 if (_request.Content == null)
                 {
                     _requestCompletionState = StreamCompletionState.Completed;
+                    if (_request.IsWebSocketH2Request())
+                    {
+                        _requestBodyCancellationSource = new CancellationTokenSource();
```

Comment:
It is used to handle request body in case of reset and exceptions, so since we use only response content stream -- this status seems reasonable to me.